### DISCUSSION TITLE
eglot has no other command

### DISCUSSION THREAD

**Discussion by azzamsa** [abougouffa/minemacs#88] [Q&A] [answered]
Hi.

Sorry to ask this question here. It should be asked in the Eglot forum. But, unlike `lsp` I can't find any other command, such `M-x eglot-code-actions` as pointed out in https://joaotavora.github.io/eglot/#index-M_002dx-eglot_002dcode_002dactions.

However, using `lsp` works as usual.

Any idea how to debug it? I believe Eglot and LSP-mode are using the same language server. So it is unlikely the language server limitation.

Thanks! :heart:

**Comment by abougouffa** [comment, marked as answer]
Not sure if I've understood your question. However, as `eglot`'s commands like `eglot-code-actions` are not autoloaded, they won't be visible in `M-x` until the package is loaded. Try to call `eglot` then see if `M-x eglot-code-actions` shows something.

**Reply by azzamsa** [reply]
Oh, I see. Turns out it shut down so that I can't access its other method.

I am wondering why eglot doesn't start automatically in Rust files. Even if rust-mode registered here https://github.com/abougouffa/minemacs/blob/e75ba00f5eefff7e7a9ff6b4cf4c453ae36d1710/modules/me-prog.el#L76.

Do we (minemacs users) expected to set those hooks manually?

**Reply by abougouffa** [reply]
In MinEmacs, the `treesit-auto` package remaps the modes to its Tree-sitter equivalent modes, if available. So, when opening a Rust file, it opens in `rust-ts-mode` not `rust-mode`. It seems that I forgot to add `rust-ts-mode` to the list. I've just pushed a fix for this.

**Reply by azzamsa** [reply]
Cool, now it is fixed!
